Feature/98 update error message

cmcWebCode40 opened the pull request:
- Updated Validation error 

closes #98

cmcWebCode40 committed:
hofix: added vee-validate declaration module

cmcWebCode40 committed:
Merge branch 'develop'

cmcWebCode40 committed:
fix: modified error message for validation schema
#98

cmcWebCode40 committed:
fix: resolved git conflict
#98

dmty commented on src/shims-vue.d.ts:
```diff
@@ -7,3 +7,4 @@ declare module '*.vue' {
 
 
 declare module "worker-loader";
+declare module "vee-validate";
```
You probably don't need to declare the module here, vee-validate is [written in typescript](https://github.com/logaretm/vee-validate/tree/main/packages/vee-validate/src).

cmcWebCode40 replied:
removed

dmty commented on src/utils/formValidation.ts:
```diff
@@ -11,20 +11,14 @@ const parseDateString = (value: string, originalValue: string) => {
+    .min(5)
+    .label("Registration Number must be greater than or equal to 5"),
+  regExpDate: yup.date().transform(parseDateString).min(today).required(),
+  healthType: yup.string().required("Health Type"),
```
Some labels actually became less clear then before

cmcWebCode40 replied:
refactored.

aurelticot reviewed (changes requested):
There's been conflicts and overriding in here, please rework the whole `yup` object.

Look at the API reference https://www.npmjs.com/package/yup, most of the methods take an optional message parameter used when the not valid.
The `label(...)` method is not to set error messages. It should be the same label as what's displayed in the UI

cmcWebCode40 committed:
fix:refactored the yup object
-#98

cmcWebCode40 commented:
> There's been conflicts and overriding in here, please rework the whole `yup` object.
> 
> Look at the API reference https://www.npmjs.com/package/yup, most of the methods take an optional message parameter used when they not valid. The `label(...)` method is not to set error messages. It should be the same label as what's displayed in the UI

Hi @aurelticot,

I have refactored the yup object and modified the `.label()` method to only override the field name so it won't be displayed as camelCase in the UI.
also, add the custom error message for the `.min()` as the second argument .

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,21 @@ const parseDateString = (value: string, originalValue: string) => {
-  firstName: yup.string().required().label("First Name"),
-  did: yup
-    .string()
+  firstName: yup.string().trim().required().label("First name"),
```
The label is not correct, should be `First Name`

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,21 @@ const parseDateString = (value: string, originalValue: string) => {
-    .string()
+  firstName: yup.string().trim().required().label("First name"),
+  did: yup.string().required().min(50).max(50).label("Valid DID"),
+  lastName: yup.string().required().min(2).label("Last name"),
```
The label is not correct, should be `Last Name`.
How about adding a `.trim()` like for the first name?

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,21 @@ const parseDateString = (value: string, originalValue: string) => {
-  did: yup
-    .string()
+  firstName: yup.string().trim().required().label("First name"),
+  did: yup.string().required().min(50).max(50).label("Valid DID"),
```
I guess the label should be just `DID`, no?
There is a `.length()` method on `.string()` which probably better then both `.min()` and `.max()`.

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,21 @@ const parseDateString = (value: string, originalValue: string) => {
-    .label("A valid `DID` must contain 50 characters"),
-  lastName: yup.string().required().min(2).label("Last Name"),
-  regNumber: yup.number().required().min(5).label("Registration Number"),
+    .min(5, "Registration Number must be greater than or equal to 5")
```
What the displayed message looks like with this message argument? As yup uses the label in its default messages, I'm wondering if it uses it as well in the custom messages. If so the label will be duplicate because it is also in the custom message. If not then it's ok.

cmcWebCode40 replied:
I just removed the `.min()` custom  because I notice yup default message is clear , but the label does not duplicate

aurelticot replied:
Ok.
It here is still the question from the original issue when it's not a number, the default message from yup says something like "value was NaN". Can you double check if there's a way to override this default message?
Thanks

cmcWebCode40 replied:
ok, will do that shortly

cmcWebCode40 replied:
@aurelticot,

I have updated the yup schema to handle type customer error using the `.typeError()` method.

aurelticot reviewed (changes requested):
Did some inline comments for some changes

cmcWebCode40 committed:
fix:updated the yup schema validation
#98

cmcWebCode40 committed:
fix:update yup custom message
#98

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,22 @@ const parseDateString = (value: string, originalValue: string) => {
-    .label("A valid `DID` must contain 50 characters"),
-  lastName: yup.string().required().min(2).label("Last Name"),
-  regNumber: yup.number().required().min(5).label("Registration Number"),
+    .min(10000, "Registration Number must b have atleast 5  characters")
```
Can you fix the typo and change "characters" by "digits"?

cmcWebCode40 replied:
@aurelticot  do you mean characters to digits?

cmcWebCode40 committed:
fix:fixed typo in validation message
#98

aurelticot commented on src/utils/formValidation.ts:
```diff
@@ -11,19 +11,22 @@ const parseDateString = (value: string, originalValue: string) => {
-    .label("A valid `DID` must contain 50 characters"),
-  lastName: yup.string().required().min(2).label("Last Name"),
-  regNumber: yup.number().required().min(5).label("Registration Number"),
+    .min(10000, "Registration Number must have atleast 5 digits")
```
Still a typo

cmcWebCode40 replied:
@aurel Please can you help me identify the typo you are referring to?

aurelticot replied:
"... at least ..."

cmcWebCode40 replied:
Fixed

cmcWebCode40 committed:
fix: fixed typo in yup schema
#98

aurelticot reviewed (approved):
Ok
